src/c++/perf_analyzer/data_loader.cc

```diff
@@ -633,4 +601,81 @@ DataLoader::ReadTextFile(
   return cb::Error::Success;
 }
 
+cb::Error
+DataLoader::ValidateTensor(
+    const ModelTensor& model_tensor, const int stream_index,
+    const int step_index)
+{
+  std::string key_name(
+      model_tensor.name_ + "_" + std::to_string(stream_index) + "_" +
+      std::to_string(step_index));
+
+  int element_count;
+  int64_t batch1_byte;
+
+  auto data_it = input_data_.find(key_name);
+  if (data_it == input_data_.end()) {
+    data_it = output_data_.find(key_name);
+  }
+  if (data_it == output_data_.end()) {
+    return cb::Error("Can't validate a nonexistent tensor");
+  }
+
+  auto shape_it = input_shapes_.find(key_name);
+  if (shape_it != input_shapes_.end()) {
+    element_count = ElementCount(shape_it->second);
+    batch1_byte = ByteSize(shape_it->second, model_tensor.datatype_);
+  } else {
+    element_count = ElementCount(model_tensor.shape_);
+    batch1_byte = ByteSize(model_tensor.shape_, model_tensor.datatype_);
+  }
+
+  // Validate that the shape is specified if variable-sized
```

Comment:
You have 3 commented and clearly separate chunks of code here. This looks like 3 methods to me. The method names can emcompass the comments and the comments don't risk going stale.

Reply:
Refactored